x-pack/solutions/security/plugins/cloud_security_posture/public/pages/configurations/latest_findings/latest_findings_group_renderer.tsx

```diff
@@ -45,8 +45,20 @@ export const groupPanelRenderer: GroupPanelRenderer<FindingsGroupingAggregation>
     <NullGroup title={title} field={selectedGroup} unit={NULL_GROUPING_UNIT} />
   );
 
+  const getGroupPanelTitle = () => {
+    if (bucket.resourceName?.buckets) {
+      return (
+        <>
+          <strong>{bucket.resourceName.buckets[0]?.key}</strong> - {bucket.key_as_string}
+        </>
+      );
+    }
+
+    return <strong>{bucket.key_as_string}</strong>;
+  };
```

Comment:
note that an empty array in `bucket.resourceName?.buckets` can count as true, which will lead into undefined `bucket.resourceName.buckets[0]?.key` that will be presented in bold.

instead, i suggest to directly check that we have a `bucket.resourceName.buckets[0]?.key` to decide if we want to render it or just `bucket.key_as_string`.

i also suggest to add named constants for those values so it will be easier to understand what exactly are we rendering.

```suggestion
const getGroupPanelTitle = () => {
  const resourceId = bucket.resourceName?.buckets[0]?.key;
  const resourceName = bucket.key_as_string || 'Some Default Value';

  return resourceId ? (
    <>
      <strong>{resourceId}</strong> - {resourceName}
    </>
  ) : (
    <strong>{resourceName}</strong>
  );
};
```

Reply:
@JordanSh fixed, although i didnt use resourceId variable as it is not always resourceId.

Reply:
can you elaborate? what else can it be?

Reply:
@JordanSh account id for example. any category from the group by menu.